### DISCUSSION TITLE
Can't change color scheme on my Mac's safari

### DISCUSSION THREAD

**Discussion by WindowsXp-Beta** [wangchucheng/hugo-eureka#54] [Q&A] [answered]
It can't change color mode on my Mac's safari.
When I press the `light` button for example , there's an error occur.
```
TypeError: window.matchMedia("(prefers-color-scheme: dark)").addEventListener is not a function. (In 'window.matchMedia("(prefers-color-scheme: dark)").addEventListener('change',switchDarkMode)', 'window.matchMedia("(prefers-color-scheme: dark)").addEventListener' is undefined)
switchMode — eureka.min.js:45
（匿名函数） — sjtu-windowsxp.github.io:156
switchMode — eureka.min.js:34:97
```
I'm not familiar with javascript so I don't know how to fix this error.
However, I'm sure it's caused by safari because it works well on Firefox and safari on my iPhone and iPad.
Can I fix this problem by modifying the source code?
By the way, My MacOS version is Catalina 10.15.7

**Comment by wangchucheng** [comment]
I tested it in an Apple DTK. The color schema works well on its safari. Can this problem be reproduced?

**Reply by WindowsXp-Beta** [reply]
Thanks.
I just updated my safari and it works.
I'll check if my browser is the latest version next time.

**Comment by WindowsXp-Beta** [comment, marked as answer]
Update the safari solves this problem.
